src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -286,6 +286,36 @@ public void EnsureRecordingTest()
             }).Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void EnsureRecordingClearedTest()
+        {
+            RemoteExecutor.Invoke(() => {
+                Activity.ForceDefaultIdFormat = true;
+                Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+
+                ActivitySource aSource = new ActivitySource("EnsureRecordingTest");
+
+                ActivityListener listener = new ActivityListener
+                {
+                    ShouldListenTo = (activitySource) => true,
+                    Sample = (ref ActivityCreationOptions<ActivityContext> activityOptions) => ActivitySamplingResult.PropagationData
```

Comment:
>ActivitySamplingResult.PropagationData [](http://example.com/codeflow?start=95&length=38)

Could you add extra case setting returning `ActivitySamplingResult.AllDataAndRecorded` and ensuring the `Recorded` is set in the created activity?

Reply:
Updated. There should be 3 sampling tests now. One for each mode: AllDataAndRecorded, AllData, & PropagationData. Tests verify `IsAllDataRequested` & `Recorded` match what is expected given the sampling decision. LMK if you want more.